pkg/controllers/provisioning/provisioner.go

```diff
@@ -343,10 +343,25 @@ func (p *Provisioner) Schedule(ctx context.Context) (scheduler.Results, error) {
 	}
 	// Mark in memory when these pods were marked as schedulable or when we made a decision on the pods
 	p.cluster.MarkPodSchedulingDecisions(results.PodErrors, pendingPods...)
+	p.MarkPodSchedulingDecisionsNodeRegistrationHealthy(ctx, results)
 	results.Record(ctx, p.recorder, p.cluster)
 	return results, nil
 }
 
+// MarkPodSchedulingDecisionsNodeRegistrationHealthy iterates through the nodeClaims
+// and existing nodes and marks podHealthyNodePoolScheduledTime time for pods
+func (p *Provisioner) MarkPodSchedulingDecisionsNodeRegistrationHealthy(ctx context.Context, results scheduler.Results) {
+	now := time.Now()
+	for _, n := range results.NewNodeClaims {
+		p.cluster.UpdatePodHealthyNodePoolScheduledTime(ctx, n.Labels[v1.NodePoolLabelKey], now, n.Pods...)
```

Comment:
nit: Do we need to pass the time through? If we are marking it, can we just assume that we are talking about `time.Now()`